### DISCUSSION TITLE
opentrack 2023.3.0

### DISCUSSION THREAD

**Discussion by sthalik** [opentrack/opentrack#1746] [Announcements]
### Features

- `tracker/tobii`: New tracker by @khoanguyen-3fc.
- `tracker/neuralnet`: Pre-release version 0.2 of the neural network by @DaWelter is available in settings.
- `tracker/neuralnet`: Fix inverted roll sign (by @DaWelter, reported by @ddzzkunye).
- `tracker/hatire`: Add non-standard higher baud rates (by @sthalik, explained by @GO63-samara).
- `i18n`: Improve the Chinese translation (by @ddzzkunye).
- `contrib`: Add libusbK driver kit [installer](https://github.com/opentrack/opentrack/blob/master/contrib/libusbK-dev-kit/libusbK-3.1.0.0-setup.exe). It's not being run by itself.

### Fixes

- `filter/ewma`: Fix wrong background color in the filter settings dialog (by @ddzzkunye and @sthalik).

<hr /><em>This discussion was created from the release <a href='https://github.com/opentrack/opentrack/releases/tag/opentrack-2023.3.0'>opentrack 2023.3.0</a>.</em>

**Comment by Norhiro** [comment]
For Fedora, if you want to use nueuralnet, you need one more depandance "onnxruntime-devel".

**Comment by pavlovRide** [comment]
Can you please add .dmg file for Mac?

**Comment by Henparty** [comment]
Just writing to say thanks as ever to everyone who works on this. Merry Xmas and NYs :)

**Comment by virgis21** [comment]
Great software! I own Mac M1 and can not find the latest version compiled for Mac, can someone do so? :) Thank you very much!

**Comment by Abulfaz2022** [comment]
Hello ! I use Tobii eye Tracker 5 as input and mouse emulation as output. This works great in combination, but as soon as I start the game Snowrunner, it all stops working.((Help with advice on how to fix this. Thanks in advance for the answer. By the way, tobii eye tracker 5 works in the snowrunner game if you use the Project İris program But it is not suitable because the camera speed is prohibitive and there is no way to customize it and there is no support for HeadTracking only gaze.

**Reply by khoanguyen-3fc** [reply]
Could you provide more information? Have you noticed any movement of the mouse on the game's main screen? Also, did moving your physical mouse have any effect?

**Reply by Abulfaz2022** [reply]
The mouse itself works fine in Snowrunner if I physically move it. That is, it’s not about the mouse. I understand why you’re asking this. And it’s not about Tobii Eye Tracker 5. It also works in this game if you use the Project Iris program, which, as I wrote above, is not suitable for me. That is, the issue here is not in the devices; there is a high probability that the issue is in Opentrack, or, more correctly, in connection with the specific game Snowrunner. It happens this way. I run Opentrack in conjunction with Tobbi and the mouse, after which the octopus begins to move and the mouse cursor clearly follows the change in the position of my head. But as soon as I run Snowrunner even in windowed mode, the mouse stops responding to head movements.I checked it on two monitors. Snowrunner was running on one, on the other a desktop window with Opentrack. The octopus moves but there is zero reaction in the game. I hope I was able to convey the information in detail, what is the essence of the problem?

**Reply by Abulfaz2022** [reply]
İ have just solve this problem ) Just run opentrack as administrator and change mouse protocol to direct movement and filter to NaturalMovement  .So its start to work )Maybe someone find this useful

**Reply by khoanguyen-3fc** [reply]
It's good to hear that you've solved the problem. However, it's not recommended to run any program as an administrator unless it's necessary. It looks like the game was accidentally run as an administrator.

**Reply by Abulfaz2022** [reply]
It’s not entirely clear how the global center key works. Let’s say I need the cursor to move to the center of the screen when this global shortcut key is pressed. But instead, the cursor ends up in the most unpredictable parts of the screen. How can I fix this?

**Comment by Abulfaz2022** [comment]
It’s not entirely clear how the global center key works. Let’s say I need the cursor to move to the center of the screen when this global shortcut key is pressed. But instead, the cursor ends up in the most unpredictable parts of the screen. How can I fix this?

**Reply by sthalik** [reply]
It's not really possible to center the view in games with mouselook from opentrack's side.

**Comment by DurgeDriven** [comment]
Thanks so much for all your work.

Please add Le Mans Ultimate

Thank You

**Reply by sthalik** [reply]
Isn't it enough that it shows as an unknown game? It should still work as-is.

**Comment by lukakrzatala** [comment]
Hello, great work. Head tracking is fantastic, trackIr goes to closet.
I have question: is there any possibility that view will return to center if tracked item (In my case my head :) goes out of range? Maybe there is some option for that?

Thanks!

Edit.
I was strugilling about that for few years. (Ok it wasn so big issue, but sometimes, annoying) Now when I asked for that I was thinking about it more,. and....OMG! it so simple! xD I ve Just cut the end of scale, so tracking will go back on center, when it lost.

**Reply by Keebler750** [reply]
What do you mean 'cut the end of scale?' .... this sounds like a possible solution to something I've been struggling with.

**Reply by lukakrzatala** [reply]
Lol. I came here because of the new message in topic and mail notificiation — sorry for my late reply  i missed your question. But  look at this coincidence It's  exactly a year today! :D.
About your question: You need to set up tracking with a top movement limit. So if you turn your head too much, the tracking stops. As far as I remember, it automatically re-centers itself afterward. Sorry, I'm currently using VR, so I can only rely on my memory.

**Comment by Nyluss** [comment]
Is there a way to set opentrack to a different vjoy device? I have 2 devices that are set to vjoy 1 and vjoy 2 and would like to set opentrack to use vjoy 3. Is this possible?

**Comment by SGTDRE** [comment]
I just want to say thank you for making this, it is working well with Train Sim World 4, have not tried it in TSW3 or TSW2 yet, but With Tobii 5 and Opentrack I'm still trying to get the hang of it all figure everything out, but for the most part, it is working well. 
I do have a question though, is there a way to get OpenTrack working with the following games and more and Tobii 5 
SubwaySim Hamburg
# 693555 Steam DB 
Railroader
# 598959 Steam DB

**Comment by elimumm** [comment]
I am getting this message: 

_Error occurred while loading protocol Microsoft FSX SimConnect

Install FSX/Prepar3D SimConnect SDK._

I downloaded Prepar3D_v6_SDK_Setup_6.1.10.31609 and MSFS_SDK_Core_Installer_0.24.3.0 . Still no luck. I am windows 11 and I believe I downloaded MSFS off the xbox/windows store on my PC.

**Comment by Supergabry13th** [comment]
Hey I can't see neuralnet in the input menu, where can I find it?

**Comment by Metal-o-Holic** [comment]
Does opentrack work with trackIR

**Reply by lukakrzatala** [reply]
Yes. It works fine.

**Reply by Metal-o-Holic** [reply]
awsome  can´t wait to try it out

**Reply by Metal-o-Holic** [reply]
is there any guide how to set it up ?
I quickly glanced the trackir and open ir settings. I would assume track ir is suppose to be selected as input on the open track. But there is no option named ”Track ir”

**Comment by B14Ape** [comment]
Hi good folk of OpenTrack! 
Great work with the software, I use it for Farming Simulator 25 and Opentrack ability to use the Z axis is brilliant (which for some reason doesn't work with Tobii Eyetracker 5 and the in built game software). I have a question, using OpenTrack instead of the in built software disables the gaze function. Is there a way to enable this as well as the Z axis?

**Reply by Henparty** [reply]
"Beam Eye Tracker" is separate software ($30 in Steam) that uses Opentrack to do eyetracking with a regular webcam. I tried their timed demo and didn't think it was worth that but ymmv

**Comment by andydevine-stack** [comment]
First let me say thanks for all the work you guys do on this!  Much appreciated!  I got sick of breaking the TIR stuff and it failing over time.  Got myself the Delan stuff and it works great!  It's been years now.  I used FaceTrack no IR for years and now I'm on OpenTrack.  Thanks again!
Windows 11 24H2 (Not 25H2)

I am getting the following error **_periodically_** when using a batch file to start the software I use to run my Flight Sim  BMS Falcon 4.38 (yay!)

DEBUG [F:\dev\opentrack\api/plugin-support.hpp:106]: library "kinect-face" failed: "Cannot load library C:\\Program Files (x86)\\opentrack\\modules\\\\opentrack-tracker-kinect-face.dll: The specified module could not be found."

This error  **_never_** happens when I simply use the Windows 11 UI to open OT.....only when opened with the batch file (Run as Administrator checked)

Once in a while, usually after a reboot, but not consistently reproduceable, I get the above CMD Prompt DEBUG window telling me it cannot find Microsoft Kinect module.  Here's what I've done to troubleshoot and solve the problem:

- Checked the box for enable tray (without Minimize on Enable checked)  Behaves a little weirdly in the tray/Taskbar but that's not my problem.  The Debug did NOT reappear....until after reboot.
- Checked boxes for both Enable Tray and Minimize on Enable (still exhibits weird behavior in the Tray/Taskbar; not my problem.
- Unchecked both boxes for both (back to where I started)
- Input the Modules Folder location into the Custom Location under FreeTrack 2.0 Enhanced Library Locations (got the best results after trying this)  It more consistently did NOT come up with the DEBUG Screen

After I input the Custom Location under FreeTrack 2.0 Enhanced settings>Library Locations, it seemed to do well and not popup after testing a dozen starts and kills using the batch files.  Then I reboot and BAM!  The DOS DEBUG screen comes back!  I just want that to go away.  I don't know what to do about it. and what's with all the \\ marks in the debug file location?  Could that be why it can't find it.??

Pardon my ignorance.  I only know enough to create problems not solve them......ugh.  sorry.

Maybe there's a command line argument/switch I can use to NOT look for Kinect Face api?  It's the only one I get.  Kinect Face....

I know, I know, Just use the shortcut on the desktop and stop fooling around.......

**Reply by GO63-samara** [reply]
If you are not using the Kinect tracker, simply delete the file opentrack\modules\opentrack-tracker-kinect-face.dll

**Reply by andydevine-stack** [reply]
Thank you Samara-G!  I'll give it a try.

And it will not look for it then?  Just curious because that sounds counter intuitive.....delete what it's looking for....but maybe OT is looking because the module *Kenect*.dll is already in there?
